### DISCUSSION TITLE
Add `let x in r = ...` short hand?

### DISCUSSION THREAD

**Discussion by magnus-madsen** [flix/flix#8581] [Ideas]
Consider the expression:

```scala
region rc {
    let _a = Array#{Array#{1} @ rc, Array#{2} @ rc, Array#{3} @ rc} @ rc; 
    ()
}
```

Feels pretty bad, right?

Maybe can we add syntactic sugar that compiles to the above, but is written as:

```
region rc {
    let _a in rc = Array#{Array#{1}, Array#{2}, Array#{3}}; 
    ()
}
```

In the simplest version, we do something like the following:

- We make the region expr optional for array literals etc. etc.
- We add `let ... in` and new type of implicit environment.
- While visiting array literals etc. without an explicit region, we automatically fill it in.

**Comment by magnus-madsen** [comment]
A criticism is: Why only do this for let?

**Reply by JonathanStarup** [reply]
yeah id say we should explore more general solutions. seems very related to implicit arguments
